### PR TITLE
Implementation of the scheduling auction client and host functionality

### PR DESCRIPTION
**Pull request by autodidaddict**
Functionality for holding actor and provider "auctions".

An auction basically asks the entire lattice, "can you start the given entity with the given constraints?". The answer will be acknowledged by a host if:

* That entity is not already running on the host (no duplicates of a single entity are allowed within the same host)
* The constraints specified match the host's labels

### CODE REVIEW

File: crates/wasmcloud-host/src/control_interface/ctlactor.rs [hunk 1]
```diff
@@ -55,12 +55,12 @@ impl Actor for ControlInterface {
...
-    fn handle(&mut self, msg: PublishEvent, _ctx: &mut Context<Self>) -> Self::Result {
-        /*if self.client.is_none() {
-        //    trace!("Skipping control interface event (interface disabled)");
+    fn handle(&mut self, _msg: PublishEvent, _ctx: &mut Context<Self>) -> Self::Result {
+        if self.client.is_none() {
+            //    trace!("Skipping control interface event (interface disabled)");
             return Box::pin(async move {}.into_actor(self));
         }
-        let evt = msg
+        let evt = _msg
```
Comment (brooksmtownsend) [inline crates/wasmcloud-host/src/control_interface/ctlactor.rs:63]: Is the `_` prefix intentional here on msg? I thought that was usually reserved for denoting that you wouldn't use that variable

File: tests/control.rs [hunk 2]
```diff
@@ -134,3 +134,98 @@ pub(crate) async fn basics() -> Result<()> {
 
     Ok(())
 }
+
+pub(crate) async fn auctions() -> Result<()> {
```
Comment (brooksmtownsend) [inline tests/control.rs:138]: Can't express enough how useful these tests are for understanding the code. This is 💯  for consumers of this project

**Review by brooksmtownsend — CHANGES_REQUESTED** [review]
One nitpick and LGTM

File: crates/wasmcloud-host/src/control_interface/ctlactor.rs (hunk 1 above)
Comment (autodidaddict) [reply crates/wasmcloud-host/src/control_interface/ctlactor.rs:63]: Hah, this looks like an "IDE refactor". I'll take care of it.

**Comment by autodidaddict** [comment]
@brooksmtownsend took care of that one variable name you pointed out.

**Review by brooksmtownsend — APPROVED** [review]

**Pull request merged**